packages/datadog-plugin-redis/test/legacy.spec.js

```diff
@@ -99,36 +99,41 @@ describe('Plugin', () => {
           client.stream.destroy()
         })
 
-        it('should handle errors', done => {
-          const assertError = () => {
-            if (!error || !span) return
-
-            try {
-              expect(span.meta).to.have.property(ERROR_TYPE, error.name)
-              expect(span.meta).to.have.property(ERROR_MESSAGE, error.message)
-              expect(span.meta).to.have.property(ERROR_STACK, error.stack)
-              expect(span.meta).to.have.property('component', 'redis')
-              expect(span.metrics).to.have.property('network.destination.port', 6379)
-              done()
-            } catch (e) {
-              done(e)
+        describe('known flakey test', () => {
+          this.retry(5)
+
+          // sometimes the span is missing
+          it('should handle errors', done => {
+            const assertError = () => {
+              if (!error || !span) return
```

Comment:
this test normally times out, right? We could use to chance to log _what_ is missing for the test to fail: is it `error` or is it `span`? This might give us a glimpse into what's happening

Reply:
I checked locally and what is happening basically, is that something like 20% of the time, the span is never received by the test agent.